apps/internal/oauth/resolvers.go

```diff
@@ -46,9 +46,6 @@ func newAuthorityEndpoint(rest *ops.REST) *authorityEndpoint {
 
 // ResolveEndpoints gets the authorization and token endpoints and creates an AuthorityEndpoints instance
 func (m *authorityEndpoint) ResolveEndpoints(ctx context.Context, authorityInfo authority.Info, userPrincipalName string) (authority.Endpoints, error) {
-	if authorityInfo.AuthorityType == ADFS && len(userPrincipalName) == 0 {
-		return authority.Endpoints{}, errors.New("UPN required for authority validation for ADFS")
-	}
```

Comment:
CC: @sarathys since these are his changes

Reply:
Looking at the current implementation, I think we have bugs here and need more test coverage. The UPN is apparently supposed to serve two purposes but in practice does not:
1. Determining the OID config endpoint. The helper that gets the endpoint takes a UPN argument but doesn't use it; see [here](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/16c062c187ac55067ece02de9fe65f2f7b4033cd/apps/internal/oauth/resolvers.go#L125).
2. Retrieving cached endpoints. Again, the UPN is irrelevant; see [here](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/16c062c187ac55067ece02de9fe65f2f7b4033cd/apps/internal/oauth/resolvers.go#L88-L97), and note the function returns the first cache entry matching the authority whether it matches the UPN's domain or not.

That said, I don't know how this is supposed to work: is a UPN required for authority validation on ADFS, as the current error states? How, if at all, should the client cache ADFS endpoints?

Reply:
@chlowell - ADFS authority validation is different than AAD authority validation because there is no centralized endpoint which knows all the authroities like in AAD world. In practical terms there is no authority validation possible on ADFS. Other MSALs do not implement any such validation. 

There are 2 choices: 

- if MSAL knows the authority is ADFS (because it ends with "adfs") then it ignores the flag
- if MSAL is authority type agnostic (JS and Py are), then app developer must set the flag to false by themsevels.

Reply:
Then I think we can consider cleaning up this API and adding test coverage here part of #31.

Do you prefer one choice over the other? MSAL Go isn't authority type agnostic, but it might be possible to make it so before it officially supports ADFS.